src/models/room.ts

```diff
@@ -2054,7 +2054,9 @@ export class Room extends ReadReceipt<RoomEmittedEvents, RoomEventHandlerMap> {
         for (const rootEvent of matrixEvents) {
             timelineSet.addLiveEvent(rootEvent, {
                 duplicateStrategy: DuplicateStrategy.Replace,
-                fromCache: false,
+                // We say these events are from cache as we're abusing a single timeline to maintain events
+                // from multiple threads and as such do not want to emit Room.Timeline events for these events
+                fromCache: true,
```

Comment:
What is the impact if we do end up emitting a `Room.Timeline` event?
Reading through this I fail to get that context. If the conscequences are bad enough, we probably should add a test for it too

Reply:
Can do this in a different manner in Notifier itself, will move